cranelift/codegen/meta/src/shared/legalize.rs

```diff
@@ -223,12 +223,12 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
         );
     }
 
-    {
-        let inst = sextend.bind(I128).bind(I64);
+    for &(ty, ty_half, shift) in &[(I128, I64, 63), (I64, I32, 31)] {
```

Comment:
You can use `ty.bits()`.

Reply:
It looks like the enum variants in `shared::types` are mapped to values indicating their size, so you _could_ do `ty_half as usize - 1`, but that doesn't hold for enums `ir::types`. Writing the size here seems fine.